Source/radiation/Rad_nd.F90

```diff
@@ -9,6 +9,188 @@ module rad_nd_module
 
 contains
 
+  subroutine bextrp(lo, hi, &
+                    vlo, vhi, &
+                    f, f_lo, f_hi) &
+                    bind(C, name="bextrp")
+
+    integer,  intent(in   ) :: lo(3), hi(3)
+    integer,  intent(in   ) :: vlo(3), vhi(3)
+    integer,  intent(in   ) :: f_lo(3), f_hi(3)
+    real(rt), intent(inout) :: f(f_lo(1):f_hi(1),f_lo(2):f_hi(2),f_lo(3):f_hi(3))
+
+    integer :: i, j, k
+
+    !$gpu
+
+    do k = lo(3), hi(3)
+       do j = lo(2), hi(2)
+          do i = lo(1), hi(1)
+
+             ! Extrapolate the data in the ghost zones of the box using a one-sided difference:
+             ! f(i-1) = 2 * f(i) - f(i+1)
+             !
+             ! For corners in 2D, or edges in 3D, this is applied in both directions, yielding
+             ! f_{i-1,j-1} = 4 * f(i,j) - 2 * (f(i+1,j) + f(i,j+1)) + f(i+1,j+1)
+             !
```

Comment:
I don't think that this is conservative.  The easy way to see this is imagine that f is constant, then this fills the corner with a value 3 f, instead of f.

Usually for a conservative interpolant, the weights sum to 1.